protocol-units/bridge/chains/ethereum/src/lib.rs

```diff
@@ -0,0 +1,294 @@
+use alloy::pubsub::PubSubFrontend;
+use alloy::signers::local::PrivateKeySigner;
+use alloy_network::{Ethereum, EthereumWallet};
+use alloy_primitives::{
+	private::serde::{Deserialize, Serialize},
+	Address,
+};
+use alloy_primitives::{FixedBytes, U256};
+use alloy_provider::{
+	fillers::{ChainIdFiller, FillProvider, GasFiller, JoinFill, NonceFiller, WalletFiller},
+	Provider, ProviderBuilder, RootProvider,
+};
+use alloy_rlp::{Decodable, Encodable, RlpDecodable, RlpEncodable};
+use alloy_sol_types::sol;
+use alloy_transport::BoxTransport;
+use alloy_transport_ws::WsConnect;
+use anyhow::Context;
+use bridge_shared::types::{
+	Amount, BridgeTransferDetails, BridgeTransferId, HashLock, HashLockPreImage, InitiatorAddress,
+	RecipientAddress, TimeLock,
+};
+use bridge_shared::{
+	bridge_contracts::{
+		BridgeContractCounterpartyError, BridgeContractInitiator, BridgeContractInitiatorError,
+		BridgeContractInitiatorResult,
+	},
+	bridge_monitoring::{BridgeContractCounterpartyEvent, BridgeContractInitiatorEvent},
+};
+use keccak_hash::keccak;
+use mcr_settlement_client::send_eth_transaction::{
+	InsufficentFunds, SendTransactionErrorRule, UnderPriced, VerifyRule,
+};
+
+use std::fmt::Debug;
+use utils::{send_transaction, EthAddress};
+
+pub mod utils;
+
+const INITIATOR_ADDRESS: &str = "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266";
+const RECIPIENT_ADDRESS: &str = "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266";
+const DEFAULT_GAS_LIMIT: u64 = 10_000_000_000;
+const MAX_RETRIES: u32 = 5;
+
+type EthHash = [u8; 32];
+
+pub type SCIResult<A, H> = Result<BridgeContractInitiatorEvent<A, H>, BridgeContractInitiatorError>;
+pub type SCCResult<A, H> =
+	Result<BridgeContractCounterpartyEvent<A, H>, BridgeContractCounterpartyError>;
+
+///Configuration for the Ethereum Bridge Client
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Config {
+	pub rpc_url: Option<String>,
+	pub ws_url: Option<String>,
+	pub chain_id: String,
+	pub signer_private_key: String,
+	pub initiator_address: EthAddress,
```

Comment:
I gues we can just use the signing address here instead `let contract = AtomicBridgeInitiator::new(initiator_address.0 .0, &self.rpc_provider);`. I thought it might be useful to store the initiator address and the signer address as different fields. But maybe its redunant.